### PR TITLE
Use unique_ptr instead of auto_ptr

### PR DESCRIPTION
**Pull request by ayan-b**
`auto_ptr` is deprecated.

### CODE REVIEW

**Comment by ayan-b** [comment]
Looks like Cython in Python 2.7 only works with MSVS 2008, this it won't be compatible with C++14. Maybe we should drop Windows Python 2.7 build from the CI configuration.

**Comment by liadomide** [comment]
As tvb-library is only py3 compatible, for us it would be ok to drop Python2 support in tvb-gdist also.

**Comment by ayan-b** [comment]
Dropped Python 2.7 from CI in ee47a56.

File: setup.py [hunk 1]
```diff
@@ -54,7 +54,9 @@
         sources=["gdist.pyx"],  # Filename of Cython source
         language="c++",  # Cython create C++ source
         # Disable assertions; one is failing geodesic_mesh.h:405
-        define_macros=[('NDEBUG', 1)]
+        define_macros=[('NDEBUG', 1)],
+        extra_compile_args=['--std=c++14'],
+        extra_link_args=['--std=c++14'],
```
Comment (liadomide) [inline setup.py:59]: @bvalean Could this change break our Jenkins builds for tvb-gdist?

Comment (bogdanvalean1) [reply setup.py:59]: No. I just test them locally and both are working fine.

Comment (liadomide) [reply setup.py:59]: Perfect. Thank you for testing this!

**Comment by liadomide** [comment]
@ayan-b can we then we merge this, or would you like to add some more changes here?

**Comment by ayan-b** [comment]
> @ayan-b can we then we merge this, or would you like to add some more changes here?

It's ok to merge now.

**Review by liadomide — APPROVED** [review]

**Pull request merged**